### PR TITLE
Limit waiting-message to single- or last-chunk apdus

### LINKED ISSUE #8691: Ledger Device: "Waiting..." message prints too many times for large messages
#### Problem
Because large messages prompt multiple calls to `send_adpu()`, the "Waiting..." message is printed more than once. Unnecessary.

#### Proposed Solution
Either move the print statement out of `send_adpu()`, or add logic to limit print to 1x

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
Because large messages prompt multiple calls to `send_adpu()`, the "Waiting..." message is printed more than once. Unnecessary.

#### Summary of Changes
- Update logic to only print "Waiting..." message on single- or last-chunk apdus

Fixes #8691

### CODE REVIEW

File: remote-wallet/src/ledger.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ impl LedgerWallet {
         data: &[u8],
     ) -> Result<Vec<u8>, RemoteWalletError> {
         self.write(command, p1, p2, data)?;
-        if p1 == P1_CONFIRM {
+        if p1 == P1_CONFIRM && (p2 == 0 || p2 == P2_EXTEND) {
```
Comment (garious) [inline remote-wallet/src/ledger.rs]: This needs a comment. Your intent isn't clear. `p2 == P2_EXTEND` means:
* `(p2 & P2_EXTEND) && (p2 & P2_MORE == 0)` aka, the last of a multi-part message
* No other bits in p2 will ever be used in the future

Why not this?

```rust
// If this is the first of a multi-part message
if (p1 == P1_CONFIRM) && ((p2 & P2_EXTEND) == 0) {
```

Comment (CriesofCarrots) [reply remote-wallet/src/ledger.rs]: It doesn't seem quite correct to print the message during the first chunk of a multi-part message. If something goes wrong with subsequent sends, the the cli may not in fact be waiting for an action from the device.

Could we indicate the last of a multi-part message more categorically?

Comment (CriesofCarrots) [reply remote-wallet/src/ledger.rs]: How about this, which only assumes that the 2nd bit won't be used for anything else in the future?

File: remote-wallet/src/ledger.rs [hunk 2]
```diff
@@ -227,7 +227,10 @@ impl LedgerWallet {
         data: &[u8],
     ) -> Result<Vec<u8>, RemoteWalletError> {
         self.write(command, p1, p2, data)?;
-        if p1 == P1_CONFIRM {
+        if p1 == P1_CONFIRM
+            // If this is the last part of the message
+            && (p2 >> 1 & P2_EXTEND == 0)
```
Comment (garious) [inline remote-wallet/src/ledger.rs]: That makes my brain hurt. Separate function and at least 4 unit tests?

Comment (garious) [reply remote-wallet/src/ledger.rs]: And please assume all remaining p2 bits will be used.

**Pull request merged**